File: source/extensions/remoteconversation/java/src/main/java/com/microsoft/cognitiveservices/speech/remoteconversation/RemoteConversationTranscriptionResultImpl.java
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
package com.microsoft.cognitiveservices.speech.remoteconversation;

import com.microsoft.cognitiveservices.speech.PropertyCollection;
import com.microsoft.cognitiveservices.speech.ResultReason;
import com.microsoft.cognitiveservices.speech.transcription.ConversationTranscriptionResult;
import java.math.BigInteger;

class RemoteConversationTranscriptionResultImpl extends ConversationTranscriptionResult {
    public RemoteConversationTranscriptionResultImpl() {
        super(0);
    }

    /**
     * A string that represents the user id.
     * @return the user ID string.
     */
    @Override
    public String getUserId() {
        return this.userId;
    }

    /**
     * Returns a String that represents the conversation transcription result.
     * @return A String that represents the conversation transcription result.
     */
    @Override
    public String toString() {
        return "ResultId:" + this.getResultId()+
                " Status:" + this.getReason() +
                " UserId:" + this.getUserId() +
                " Recognized text:<" + this.getText() + ">.";
    }


    /**
     * Specifies the result identifier.
     * @return Specifies the result identifier.
     */
    @Override
    public String getResultId() {
        return this.resultId;
    }

    /**
     * Specifies reason the result was created.
     * @return Specifies reason of the result.
     */
    @Override
    public ResultReason getReason() {
        return this.reason;
    }

    /**
     * Presents the recognized text in the result.
     * @return Presents the recognized text in the result.
     */
    public String getText() {
        return this.text;
    }

    /**
     * Duration of recognized speech in 100nsec increments.
     * @return BigInteger of recognized speech in 100nsec increments.
     */
    public BigInteger getDuration() {
        return this.duration;
    }

    /**
     * Offset of recognized speech in 100nsec increments.
     * @return Offset of recognized speech in 100nsec increments.
     */
    public BigInteger getOffset() {
        return this.offset;
    }

    /**
     *  The set of properties exposed in the result.
     * @return The set of properties exposed in the result.
     */
    public PropertyCollection getProperties() {
        return propertyCollection;
    }

    public void SetAllFields(String usrId, String resId, ResultReason reason, BigInteger duration, BigInteger offset, String text, PropertyCollection collection) {
        this.userId = usrId;
        this.resultId = resId;
        this.reason = reason;
        this.duration = duration;
        this.offset = offset;
        this.text = text;
        this.propertyCollection = collection;
    }

    private String userId;
    private String resultId;
    private ResultReason reason;
    private BigInteger duration;
    private BigInteger offset;
    private String text;
    private PropertyCollection propertyCollection;

}

File: source/extensions/remoteconversation/csharp/remoteconversation/remoteconversation/Extended/RemoteConversationTranscriptionResultImpl.cs
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//

using System;
using Microsoft.CognitiveServices.Speech;
using Microsoft.CognitiveServices.Speech.Transcription;

namespace Microsoft.CognitiveServices.Speech.RemoteConversation
{
    internal class RemoteConversationTranscriptionResultImpl : ConversationTranscriptionResult
    {
        public RemoteConversationTranscriptionResultImpl(
            string text,
            string resultId,
            string userId,
            ResultReason reason,
            long offsetInTicks,
            long durationInticks,
            PropertyCollection collection
            )
            : base(IntPtr.Zero)
        {
            Text = text;
            ResultId = resultId;
            UserId = userId;
            Reason = reason;
            OffsetInTicks = offsetInTicks;
            Duration = TimeSpan.FromTicks(durationInticks);
            Properties = collection;
        }

        public override string UserId { get; }

        public override string UtteranceId { get; }

        public string toString()
        {
            return "ResultId:" + this.ResultId +
                    " Status:" + this.Reason +
                    " UserId:" + this.UserId +
                    " Recognized text:<" + this.Text + ">.";
        }

        public override string ResultId { get; }

        public override ResultReason Reason { get; }

        public override string Text { get; }

        public override TimeSpan Duration { get; }

        public override long OffsetInTicks { get; }

        public override PropertyCollection Properties { get; }
    }
}
